DeDriftAndResample/DeDriftAndResamplePipeline.sh

```diff
@@ -27,1094 +27,614 @@
+log_Msg "AtlasFolder: ${AtlasFolder}"
+
+T1wFolder="${StudyFolder}/${Subject}/T1w"
+log_Msg "T1wFolder: ${T1wFolder}"
+
+NativeFolder="${AtlasFolder}/Native"
+log_Msg "NativeFolder: ${NativeFolder}"
+
+NativeT1wFolder="${T1wFolder}/Native"
+log_Msg "NativeT1wFolder: ${NativeT1wFolder}"
+
+ResultsFolder="${AtlasFolder}/Results"
+log_Msg "ResultsFolder: ${ResultsFolder}"
+
+#Naming Conventions
+DownSampleFolderNames=""
+DownSampleT1wFolderNames=""
+for LowResMesh in ${LowResMeshes} ; do
+	DownSampleFolderNames=`echo "${DownSampleFolderNames}${AtlasFolder}/fsaverage_LR${LowResMesh}k "`
+	DownSampleT1wFolderNames=`echo "${DownSampleT1wFolderNames}${T1wFolder}/fsaverage_LR${LowResMesh}k "`
+done
+log_Msg "DownSampleFolderNames: ${DownSampleFolderNames}"
+log_Msg "DownSampleT1wFolderNames: ${DownSampleT1wFolderNames}"
+
+# Concat Reg
+log_Msg "Concat Reg"
+for Hemisphere in L R ; do
+	if [ $Hemisphere = "L" ] ; then 
+		Structure="CORTEX_LEFT"
+		DeDriftRegFile=`echo ${DeDriftRegFiles} | cut -d " " -f 1`
+	elif [ $Hemisphere = "R" ] ; then 
+		Structure="CORTEX_RIGHT"
+		DeDriftRegFile=`echo ${DeDriftRegFiles} | cut -d " " -f 2`
+	fi 
+
+	log_Msg "Hemisphere: ${Hemisphere}"
+	log_Msg "Structure: ${Structure}"
+	log_Msg "DeDriftRegFile: ${DeDriftRegFile}"
+
+	if [ ! ${RegName} = ${OutputRegName} ] ; then #RegName is already the completed registration, don't overwrite
+		${Caret7_Command} -surface-sphere-project-unproject ${NativeFolder}/${Subject}.${Hemisphere}.sphere.${RegName}.native.surf.gii ${AtlasFolder}/${Subject}.${Hemisphere}.sphere.${HighResMesh}k_fs_LR.surf.gii ${DeDriftRegFile} ${NativeFolder}/${Subject}.${Hemisphere}.sphere.${OutputRegName}.native.surf.gii
 	fi
-	log_Msg "After delimiter substitution, MyelinMaps: ${MyelinMaps}"
 
-	if [ "${fixNames}" = "NONE" ] ; then
-		fixNames=()
-	else
-		#fixNames=`echo "$fixNames" | sed s/"@"/" "/g`
-		IFS=@ read -a fixNames <<< "${fixNames}"
-	fi
-	log_Msg "After delimiter substitution, fixNames: ${fixNames[@]}"
+	# Make MSM Registration Areal Distortion Maps
+	log_Msg "Make MSM Registration Areal Distortion Maps"
+	${Caret7_Command} -surface-vertex-areas ${NativeFolder}/${Subject}.${Hemisphere}.sphere.native.surf.gii ${NativeFolder}/${Subject}.${Hemisphere}.sphere.native.shape.gii
+	${Caret7_Command} -surface-vertex-areas ${NativeFolder}/${Subject}.${Hemisphere}.sphere.${OutputRegName}.native.surf.gii ${NativeFolder}/${Subject}.${Hemisphere}.sphere.${OutputRegName}.native.shape.gii
+	${Caret7_Command} -metric-math "ln(spherereg / sphere) / ln(2)" ${NativeFolder}/${Subject}.${Hemisphere}.ArealDistortion_${OutputRegName}.native.shape.gii -var sphere ${NativeFolder}/${Subject}.${Hemisphere}.sphere.native.shape.gii -var spherereg ${NativeFolder}/${Subject}.${Hemisphere}.sphere.${OutputRegName}.native.shape.gii
+	rm ${NativeFolder}/${Subject}.${Hemisphere}.sphere.native.shape.gii ${NativeFolder}/${Subject}.${Hemisphere}.sphere.${OutputRegName}.native.shape.gii
 
-	if [ "${dontFixNames}" = "NONE" ] ; then
```

Comment:
There may be legacy launch scripts using NONE for some of these.  I would be happy to remove the NONE convention from the pipelines entirely, but our users may not be quite as happy if their old launcher scripts error for what seem like silly reasons.  @glasserm ?

Reply:
Also, if we don't support NONE, these if statements should be unnecessary, as `read -a` should generate an empty array when there is no input, as desired.

Reply:
It probably comes from positional arguments.  I would probably prefer to remove it from the launcher but not the back end code for now.

Reply:
Okay, so to make the users happy with either empty string or "NONE" for the arguments like this, we will enable the if check `if [ "${ARGUMENT}" = "NONE" ] ; then`, but in the launcher script and the default value in the newopt, we should specify empty string instead of "NONE". Am I correct about this?

Reply:
That is correct.